htdocs/compta/facture/class/facture.class.php

```diff
@@ -2386,9 +2386,9 @@ function updateline($rowid, $desc, $pu, $qty, $remise_percent, $date_start, $dat
 			$this->line->date_start			= $date_start;
 			$this->line->date_end			= $date_end;
 			$this->line->total_ht			= (($this->type==self::TYPE_CREDIT_NOTE||$qty<0)?-abs($total_ht):$total_ht);  // For credit note and if qty is negative, total is negative
-			$this->line->total_tva			= $total_tva;
-			$this->line->total_localtax1	= $total_localtax1;
-			$this->line->total_localtax2	= $total_localtax2;
+			$this->line->total_tva			= (($this->type==self::TYPE_CREDIT_NOTE||$qty<0)?-abs($total_tva):$total_tva);  // For credit note and if qty is negative, total is negative
```

Comment:
Fix into function addline looks good to me. However this part seems to be here to manage corrupted data in base due to old bug ? Right ? If i am right, it should be fixed by a sql request and not into core code. For example by adding the sql that fix data into file repair.sql